pydis_site/apps/api/serializers.py

```diff
@@ -110,6 +113,16 @@ def validate(self, attrs: dict) -> dict:
         """Validate data constraints for the given data and abort if it is invalid."""
         infr_type = attrs.get('type')
 
+        active = attrs.get('active')
+        if active and infr_type in ('note', 'warning', 'kick'):
+            raise ValidationError({'active': [f'{infr_type} infractions cannot be active.']})
+
+        user = attrs.get('user')
+        if active and Infraction.objects.filter(user=user, type=infr_type, active=True).exists():
```

Comment:
How come a [`UniqueValidator`](https://www.django-rest-framework.org/api-guide/validators/#uniquevalidator) wasn't used instead? Supposedly a `ModelSerializer` does this: 

> It will automatically generate validators for the serializer, such as unique_together validators.

I wonder, does it support generating validators for unique constraints on the model?

Reply:
I've changed this logic for the `UniqueTogetherValidator` with a custom message. It works well, although it gives slightly less specific feedback:

```py
'non_field_errors': ['This user already has an active infraction of this type.']
```

However, since this is generated for a specific infraction payload, that should not really be an issue and the message is unambiguous in the context of the query.

Reply:
So you still ended up having to manually specify the validator. It did not generate it on its own?

Reply:
As far as I can tell, it does not automatically add a validator for the constraint we've added to the model. If I remove our "custom" validator, the serializer doesn't seem to catch double active infractions and we'll get an `IntegrityError` (fail late in the database) instead.

Reply:
That's fine then. I looked at the [source](https://github.com/encode/django-rest-framework/blob/master/rest_framework/serializers.py#L1535) and it seems to only work if [`unique_together`](https://docs.djangoproject.com/en/2.2/ref/models/options/#unique-together) is used.